final_task/rss_reader/converter.py

```diff
@@ -0,0 +1,120 @@
+
+from dominate.tags import div, h2, img, p, a
+from datetime import datetime
+import dominate
+import html
+import os
+import httplib2
+import socket
+
+from fpdf import FPDF
+
+
+def is_connected() -> bool:
+    try:
+        socket.create_connection(("www.google.com", 80))
+    except Exception:
+        return False
+    return True
+
+
+def pdf_convert(news, limit, path):
+    pdf = FPDF()
+    count = 0
+    h = httplib2.Http('.cache')
+    pdf.add_font('FreeSans', '', 'FreeSans.ttf', uni=True)
+    for ind, item in enumerate(news):
+        if ind + 1 > limit:
+            print('dima')
+            break
+        pdf.add_page()
+        count_on_page = -1
+        if is_connected():
+            for image in item['image']:
+                count_on_page += 1
+                if count_on_page > 4:
+                    break
+                count += 1
+                position = count_on_page * 52 + 12
+                try:
+                    response, content = h.request(image)
+                    out = open('image' + str(count) + '.jpg', 'wb')
+                    out.write(content)
+                    out.close()
+                    pdf.image('image' + str(count) + '.jpg', 130, position, 70, 50)
+                except Exception:
+                    pass
```

Comment:
It is better to use `slices` instead of having counter and comparing it with `limit` each iteration here.